### PR TITLE
Remove non required db requests

### PR DESCRIPTION
**Pull request by Loki3000**
Remove SQL requests with empty UID for guests.

### CODE REVIEW

File: lib/private/Group/Database.php [hunk 1]
```diff
@@ -202,8 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 * if the user exists at all.
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty $uid
+		if (empty($uid)) return [];
+		
 		$this->fixDI();
-
+
```
Comment (MorrisJobke) [inline lib/private/Group/Database.php]: There should be no space at the end of a line.

File: lib/private/Group/Database.php [hunk 2]
```diff
@@ -202,8 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 * if the user exists at all.
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty $uid
+		if (empty($uid)) return [];
```
Comment (MorrisJobke) [inline lib/private/Group/Database.php]: Please always use curly brackets.

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

File: lib/private/Group/Database.php [hunk 3]
```diff
@@ -202,6 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 * if the user exists at all.
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty $uid
+		if (empty($uid)) {
```
Comment (nickvergessen) [inline lib/private/Group/Database.php]: Also falsely matches user with id `'0'`

File: lib/private/User/Database.php [hunk 4]
```diff
@@ -239,6 +238,12 @@ public function checkPassword($uid, $password) {
 	 */
 	private function loadUser($uid) {
 		if (!isset($this->cache[$uid])) {
+			//guests $uid could be NULL or ''
+			if (empty($uid)) {
```
Comment (nickvergessen) [inline lib/private/User/Database.php]: Also falsely matches user with id `'0'`

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: lib/private/User/Database.php (hunk 4 above)
Comment (Loki3000) [inline lib/private/User/Database.php]: I have asked here  [https://github.com/nextcloud/server/pull/2998#discussion_r95245581](url) about $uid=='0', but has no answer. So I was suppose it's not allowed.

File: lib/private/Group/Database.php [hunk 5]
```diff
@@ -202,6 +202,11 @@ public function removeFromGroup( $uid, $gid ) {
 	 * if the user exists at all.
 	 */
 	public function getUserGroups( $uid ) {
+		//guests has empty or null $uid
+		if ($uid===null || $uid==='') {
```
Comment (MorrisJobke) [inline lib/private/Group/Database.php]: please use spaces around the `===`.

File: lib/private/User/Database.php [hunk 6]
```diff
@@ -239,6 +238,12 @@ public function checkPassword($uid, $password) {
 	 */
 	private function loadUser($uid) {
 		if (!isset($this->cache[$uid])) {
+			//guests $uid could be NULL or ''
+			if ($uid===null || $uid==='') {
```
Comment (MorrisJobke) [inline lib/private/User/Database.php]: please use spaces around the `===`.

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by rullzer** [comment]
@Loki3000 could you squash your commits?

**Comment by Loki3000** [comment]
@rullzer sorry, I don't know how to do it.
According to [documentation](https://help.github.com/articles/about-pull-request-merges/) it could be made only by repository owner.

**Comment by rullzer** [comment]
@Loki3000 see for example http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Review by rullzer — APPROVED** [review]
Ah lets just get this in actually! THNX A LOT @Loki3000

**Pull request merged**